doc/types.go

```diff
@@ -20,6 +20,15 @@
 
 package doc
 
+// ReservedFieldName signifies the field names which are reserved for internal
+// use.
+type ReservedFieldName string
```

Comment:
Can we get away with needing to used reserved fields?  Can't we simply store properties somewhere else than the fields themselves if we have special need for them?

Reply:
Perhaps we should update the comment, because these fields aren't really internal fields. The user can, and is in fact expected, to use the ID field as well. It's just that this field has a significance attached to it (each ID field uniquely determines a document) that other fields do not have.

Reply:
Yeah this isn't used/needed atm. Will remove for now.